src/utils/misc.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Util method for determining if a particular URL link is a relative link that
+ * should point to a resource on the same site.
+ */
+export const isLinkInternal = (url: string) => {
+  const tempLink = document.createElement("a")
+  tempLink.href = url
+  return tempLink.hostname === window.location.hostname
```

Comment:
wait but these links are in reference to the user site no? 
eg. `/images/blah.png`, which doesnt belong in the cms? 

am actually ok with this cuz the edge case here is rare, but asking this for sake of completeness

Reply:
No it shouldn't be in reference to the user's site. We are not trying to determine if the image is from the user's site or not here, but rather we are only determining if a particular image link is a relative or absolute URL. When the HTML is generated from `marked.parse`, because the user may have loaded images using either a relative or absolute URL. This function is then required to determine if the image is a relative URL.

There is no loading of images right at this step, which is why the translation of the mediaSrc is necessary, and that is done at a later stage once we get the correct URLs/base64 from the backend.